### ISSUE TITLE
Filtering by Activity Status in views can cause a fatal error if using 'is none of' or 'is all of'

### ISSUE THREAD

**Issue by vingle** [eileenmcnaughton/civicrm_entity#293]
Steps to recreate:

- Create a view of CiviCRM activities
- Add a filter for Activity Status
- Select 'is none of' or 'is all of'
- Try to save, it just hangs

The errors in dblog seem to vary depending on config. On my setup it gives the below error repeatedly, filling up the watchdog table of the database to 100,000s of lines until the server is brought down. When @KarinG tested it - it gave a different error and only once, but the view also went to a white screen on saving.

```
Notice: Trying to get property 'leftTable' of non-object in Drupal\views\ManyToOneHelper->addTable() (line 87 of /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ManyToOneHelper.php)

#0 /var/www/vhosts/host/httpdocs/web/core/includes/bootstrap.inc(600): _drupal_error_handler_real(8, 'Trying to get p...', '/var/www/vhosts...', 87, Array)
#1 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ManyToOneHelper.php(87): _drupal_error_handler(8, 'Trying to get p...', '/var/www/vhosts...', 87, Array)
#2 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ManyToOneHelper.php(220): Drupal\views\ManyToOneHelper->addTable(Object(stdClass), 'civicrm_activit...')
#3 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/Plugin/views/filter/ManyToOne.php(123): Drupal\views\ManyToOneHelper->ensureMyTable()
#4 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ManyToOneHelper.php(260): Drupal\views\Plugin\views\filter\ManyToOne->ensureMyTable()
#5 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/Plugin/views/filter/ManyToOne.php(136): Drupal\views\ManyToOneHelper->addFilter()
#6 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/Plugin/views/filter/InOperator.php(390): Drupal\views\Plugin\views\filter\ManyToOne->opHelper()
#7 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ViewExecutable.php(1373): Drupal\views\Plugin\views\filter\InOperator->query(false)
#8 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ViewExecutable.php(1262): Drupal\views\ViewExecutable->_build('filter')
#9 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ViewExecutable.php(1391): Drupal\views\ViewExecutable->build(NULL)
#10 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ViewExecutable.php(1454): Drupal\views\ViewExecutable->execute(NULL)
#11 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/Plugin/views/display/DisplayPluginBase.php(2470): Drupal\views\ViewExecutable->render()
#12 /var/www/vhosts/host/httpdocs/web/core/modules/views/src/ViewExecutable.php(1662): Drupal\views\Plugin\views\display\DisplayPluginBase->preview()
#13 /var/www/vhosts/host/httpdocs/web/core/modules/views_ui/src/ViewUI.php(605): Drupal\views\ViewExecutable->preview('page_1', Array)
#14 /var/www/vhosts/host/httpdocs/web/core/modules/views_ui/src/ViewPreviewForm.php(62): Drupal\views_ui\ViewUI->renderPreview('page_1', Array)
#15 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Entity/EntityForm.php(149): Drupal\views_ui\ViewPreviewForm->form(Array, Object(Drupal\Core\Form\FormState))
#16 /var/www/vhosts/host/httpdocs/web/core/modules/views_ui/src/ViewFormBase.php(41): Drupal\Core\Entity\EntityForm->buildForm(Array, Object(Drupal\Core\Form\FormState))
#17 [internal function]: Drupal\views_ui\ViewFormBase->buildForm(Array, Object(Drupal\Core\Form\FormState), 'page_1')
#18 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Form/FormBuilder.php(532): call_user_func_array(Array, Array)
#19 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Form/FormBuilder.php(371): Drupal\Core\Form\FormBuilder->retrieveForm('view_preview_fo...', Object(Drupal\Core\Form\FormState))
#20 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Form/FormBuilder.php(629): Drupal\Core\Form\FormBuilder->rebuildForm('view_preview_fo...', Object(Drupal\Core\Form\FormState), Array)
#21 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Form/FormBuilder.php(321): Drupal\Core\Form\FormBuilder->processForm('view_preview_fo...', Array, Object(Drupal\Core\Form\FormState))
#22 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Controller/FormController.php(91): Drupal\Core\Form\FormBuilder->buildForm(Object(Drupal\views_ui\ViewPreviewForm), Object(Drupal\Core\Form\FormState))
#23 /var/www/vhosts/host/httpdocs/web/core/modules/layout_builder/src/Controller/LayoutBuilderHtmlEntityFormController.php(39): Drupal\Core\Controller\FormController->getContentResult(Object(Symfony\Component\HttpFoundation\Request), Object(Drupal\Core\Routing\RouteMatch))
#24 [internal function]: Drupal\layout_builder\Controller\LayoutBuilderHtmlEntityFormController->getContentResult(Object(Symfony\Component\HttpFoundation\Request), Object(Drupal\Core\Routing\RouteMatch))
#25 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(123): call_user_func_array(Array, Array)
#26 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/Render/Renderer.php(573): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#27 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(124): Drupal\Core\Render\Renderer->executeInRenderContext(Object(Drupal\Core\Render\RenderContext), Object(Closure))
#28 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/EventSubscriber/EarlyRenderingControllerWrapperSubscriber.php(97): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array)
#29 /var/www/vhosts/host/httpdocs/vendor/symfony/http-kernel/HttpKernel.php(151): Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}()
#30 /var/www/vhosts/host/httpdocs/vendor/symfony/http-kernel/HttpKernel.php(68): Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object(Symfony\Component\HttpFoundation\Request), 1)
#31 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/StackMiddleware/Session.php(57): Symfony\Component\HttpKernel\HttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#32 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php(47): Drupal\Core\StackMiddleware\Session->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#33 /var/www/vhosts/host/httpdocs/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(106): Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#34 /var/www/vhosts/host/httpdocs/web/core/modules/page_cache/src/StackMiddleware/PageCache.php(85): Drupal\page_cache\StackMiddleware\PageCache->pass(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#35 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php(47): Drupal\page_cache\StackMiddleware\PageCache->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#36 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php(52): Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#37 /var/www/vhosts/host/httpdocs/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#38 /var/www/vhosts/host/httpdocs/web/core/lib/Drupal/Core/DrupalKernel.php(708): Stack\StackedHttpKernel->handle(Object(Symfony\Component\HttpFoundation\Request), 1, true)
#39 /var/www/vhosts/host/httpdocs/web/index.php(19): Drupal\Core\DrupalKernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#40 {main}
```

**Comment by jackrabbithanna** [comment]
Issue is replicable, created new issue in D.O issue queue: https://www.drupal.org/project/civicrm_entity/issues/3391198
Closing this issue, in favor of that.
